File: android/src/main/java/expo/modules/gaodemap/overlays/ClusterViewModule.kt
package expo.modules.gaodemap.overlays

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

/**
 * Cluster 视图 Module
 */
class ClusterViewModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ClusterView")

    View(ClusterView::class) {
      Events("onPress", "onClusterPress")
      
      Prop<List<Map<String, Any>>>("points") { view: ClusterView, points ->
        view.setPoints(points)
      }
      
      Prop<Int>("radius") { view: ClusterView, radius ->
        view.setRadius(radius)
      }
      
      Prop<Int>("minClusterSize") { view: ClusterView, size ->
        view.setMinClusterSize(size)
      }
    }
  }
}

File: ios/overlays/ClusterViewModule.swift
import ExpoModulesCore

public class ClusterViewModule: Module {
    public func definition() -> ModuleDefinition {
        Name("ClusterView")
        
        View(ClusterView.self) {
            Events("onPress", "onClusterPress")
            
            Prop("points") { (view: ClusterView, points: [[String: Any]]) in
                view.setPoints(points)
            }
            
            Prop("radius") { (view: ClusterView, radius: Int) in
                view.setRadius(radius)
            }
            
            Prop("minClusterSize") { (view: ClusterView, size: Int) in
                view.setMinClusterSize(size)
            }
        }
    }
}